### PR TITLE
show the FDE "unavailable-reason" from snapd to the user

### PR DESCRIPTION
**Pull request by mwhudson**

### CODE REVIEW

File: subiquity/ui/views/filesystem/guided.py [hunk 1]
```diff
@@ -155,12 +155,12 @@ class TPMChoice:
             enabled=False, default=False,
             help=_("The model being installed prefers but does not require "
                    "TPM backed full-disk encryption and it is not available "
-                   "on this device")),
+                   "on this device (the reason given was \"{reason}\")")),
```
Comment (ogayot) [inline subiquity/ui/views/filesystem/guided.py:158]: minor: Full stop for consistency with other similar messages?

File: subiquity/ui/views/filesystem/guided.py [hunk 2]
```diff
@@ -155,12 +155,12 @@ class TPMChoice:
...
         StorageSafety.PREFER_UNENCRYPTED: TPMChoice(
             enabled=False, default=False,
             help=_("The model being installed does not prefer TPM backed "
                    "full-disk encryption and it is not available on this "
-                   "device.")),
+                   "device (the reason given was \"{reason}\").")),
```
Comment (ogayot) [inline subiquity/ui/views/filesystem/guided.py:163]: Minor:
As a non-native speaker, I find it confusing whether the reason given applies to:
* FDE not being available on this device
* The model _not_ preferring TPM backed FDE

but an example would probably make it clear, so it's just a potential obstacle for translators I guess :)

**Review by ogayot — APPROVED** [review]

File: subiquity/ui/views/filesystem/guided.py (hunk 2 above)
Comment (mwhudson) [reply subiquity/ui/views/filesystem/guided.py:163]: The messages are not especially likely to be human friendly (which is why they were not shown originally) but here is the example from the sample data:

image

In this case at least, I think it's reasonably clear that this is a reason for FDE not being available. I can add a note for the translator.

File: subiquity/ui/views/filesystem/guided.py (hunk 1 above)
Comment (mwhudson) [reply subiquity/ui/views/filesystem/guided.py:158]: :+1:

**Review by dbungert — APPROVED** [review]
Those are some ugly messages, but better to show them then to add another reason to send someone to the logs.  Thanks!

**Pull request merged**